tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenTestFactory.cs

```diff
@@ -30,12 +30,19 @@ public class JsonDrivenTestFactory
         private readonly string _collectionName;
         private readonly Dictionary<string, object> _objectMap;
         private readonly IJsonDrivenTestRunner _testRunner;
+        private readonly EventCapturer _eventCapturer;
 
         // public constructors
         public JsonDrivenTestFactory(IMongoClient client, string databaseName, string collectionName, string bucketName, Dictionary<string, object> objectMap)
             : this(null, client, databaseName, collectionName, bucketName, objectMap)
         {
         }
+        
+        public JsonDrivenTestFactory(IMongoClient client, string databaseName, string collectionName, string bucketName, Dictionary<string, object> objectMap, EventCapturer eventCapturer)
```

Comment:
I think, that it's better to provide `eventCapturer` instance to tests via `objectMap` dictionary.

@rstam , @vincentkam

Reply:
I think `objectMap` is more about parameters to be passed to the methods being tested (like sessions).

`eventCapturer` is not a parameter to the test methods, it is part of the testing infrastructure.

I would not try to use `objectMap` as the way to pass in the `eventCapturer`.